core-java-modules/core-java-string-conversions/src/main/java/com/baeldung/stringtoint/StringToIntEncapsulation.java

```diff
@@ -0,0 +1,53 @@
+package com.baeldung.stringtoint;
+
+import java.util.Optional;
+import org.apache.commons.lang3.math.NumberUtils;
+
+public class StringToIntEncapsulation {
+    
+    private StringToIntEncapsulation() {
+    }
+
+    public static Integer convertStringToIntUsingIntegerParseInt(String input){
+        try {
+            return Integer.parseInt(input);
+        } catch (NumberFormatException e) {
+            // log or handle the error
+            return null; // or Integer.MIN_VALUE, or some other default value
+        }
+    }
+
+    public static Integer convertStringToIntUsingIntegerValueOf(String input){
+        try {
+            return Integer.valueOf(input);
+        } catch (NumberFormatException e) {
+            // log or handle the error
+            return null; // or Integer.MIN_VALUE, or some other default value
+        }
+    }
+
+    public static Integer convertStringToIntUsingIntegerDecode(String input){
+        try {
+            return Integer.decode(input);
+        } catch (Exception e) {
+            // log or handle the error
+            return null; // or Integer.MIN_VALUE, or some other default value
+        }
+    }
+
+    public static Integer converStringToIntUsingOptional(String input){
+        Optional<Integer> parsedInt;
+        try {
```

Comment:
As per described in the video review, I'd rather remove this _try-catch_ block and go with this instead:

**return Optional.ofNullable(Integer.parseInt(s)).orElse(null);** 

(remember that **null** not necessarily is the best option here)

Reply:
so orElse doesn't handle the scenario when parsing fails. It throws a number format exception. It only handles scenarios where we receive a null value.